examples/dispatch_iface.yml

```diff
@@ -0,0 +1,21 @@
+---
+dispatch:
+    interface-types:
+        - name: vxcan
+          activation: "ip link add $name type vxcan peer $peer"
+          deactivation: "ip link del $name"
+          allowed-variable-names:
+              - peer
+          kernel-index-getter: "cat /sys/class/net/$name/ifindex"
+interfaces:
+    - name: vxcan1
+      type: dispatch:vxcan
+      description: "vxcan interface created by dispatch script"
+      dispatch:
+          variables:
+              peer: "vxcan1-ep"
+      ipv4:
+          enabled: true
+          address:
+              - ip: 192.0.2.1
+                prefix-length: 24
```

Comment:
I the unstructured approach at type can be replaced by "name" field under "distpatch" 
```yaml
- name: vxcan1
      type: dispatch
      description: "vxcan interface created by dispatch script"
      dispatch:
          name: vxcan
          variables:
              peer: "vxcan1-ep"
      ipv4:
          enabled: true
          address:
              - ip: 192.0.2.1
                prefix-length: 24
```

Reply:
A interface need to have a interface type.

Reply:
Can't it be `type: dispatch` ? then we just specify the distpatch type under `distpatch`

Reply:
User need to look into the `dispatch` section to know the type of this interface seems contradiction to the definition of `interface-type`.

Reply:
Well is like checking if a bridge is stp or not you look at the "bridge" section